Annotations/Classes/Views/TextCanvas.swift

```diff
@@ -8,32 +8,70 @@
 import Foundation
 import TextAnnotation
 
-protocol TextCanvas: TextAnnotationCanvas, TextViewDelegate where Self: CanvasView {
+protocol TextCanvas: TextAnnotationCanvas, TextViewDelegate, TextAnnotationDelegate where Self: CanvasView {
   var selectedItem: CanvasDrawable? { get set }
   var model: CanvasModel { get set }
 }
 
 extension TextCanvas {
-  func createTextView(origin: PointModel) -> TextView {
-    let newTextView = createTextAnnotation(text: "", location: origin.cgPoint)
+  func createTextView(text: String = "", origin: PointModel) -> TextView {
+    let newTextView = createTextAnnotation(text: text, location: origin.cgPoint)
     newTextView.delegate = self
     
-    let textModel = TextModel(origin: origin, text: "")
+    let textModel = TextModel(origin: origin,
+                              text: text)
     model.texts.append(textModel)
     
     let state = TextViewState(model: textModel, isSelected: false)
     
-    let newView = TextViewClass(state: state, modelIndex: model.texts.count - 1)
-    newView.view = newTextView
+    let modelIndex = model.texts.count - 1
     
+    let newView = TextViewClass(state: state,
+                                modelIndex: modelIndex,
+                                view: newTextView)
+    newView.delegate = self
+
     return newView
   }
+  
+  func createTextView(textModel: TextModel, index: Int) -> TextView {
+    let newTextView = createTextAnnotation(modelable: textModel)
+    
+    newTextView.delegate = self
+    
+    let state = TextViewState(model: textModel, isSelected: false)
+    
+    let newView = TextViewClass(state: state, modelIndex: index, view: newTextView)
+    newView.delegate = self
+    
+    return newView
+  }
+	
+	
+  func redrawTexts(model: CanvasModel) {
+    for (index, model) in model.texts.enumerated() {
+      let view = createTextView(textModel: model, index: index)
+      view.delegate = self
+      add(view)
+      
+      DispatchQueue.main.async {
```

Comment:
@mirkokiefer You can see that` redrawTexts()` is called in` redraw()` method of CanvasViewClass extension where the other redraws are called. There is no question about main thread here because they are all called on the main thread.
I added DispatchQueue.main.async here because there was a problem with rendering and text annotation are rendered not correctly before. This is a trick that moves the rendering to the next Main Run Loop iteration. I understand that it is hacky and tried to avoid it while working on it but didn't succeed. 
I will check it again.

Reply:
Oh you want to get it to the next cycle - maybe its enough if we at least just do it once around the loop? Of course getting rid of it altogether would be ideal.

Reply:
@mirkokiefer I meant Main Run Loop of every application. I think it works the same on Mac OS and iOS -> https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Multithreading/RunLoopManagement/RunLoopManagement.html
The idea is if you call `DispatchQueue.main.async `even from the main thread the system moves it on the next Main Run Loop iteration so some rendering is completed on the current iteration and the other one will complete on the next iteration.
This trick sometimes helps with the case as we have.
If I just remove this `DispatchQueue.main.async` redo/undo for text annotations will not be rendered. But as I said I will dive into it again and try to find how to sort this out.

Reply:
@mirkokiefer I finally find the reason and fixed it. I had to` resetFontSize()` before updating frames. This also fixed an issue with blinkings of text annotation of redo / undo.